Backend feature/update models

mtyazici opened the pull request:
Things done for this PR:
- I added two new models named customer and vendor. They inherit from client model. 
- I put the common endpoints for customer and vendor to a common router named `client`. 
- I changed the controllers, services and routers so that vendor and customer models could be accessed via our API.
- I deleted Google functionality from vendor model.

Things to be done in future:
- We should create the necessary models for fields in vendor and customer model. Right now their type are all strings.
- I updated the api-docs but they could be improved.
- We could reduce signupWithGoogle and loginWithGoogle to only SigninWithGoogle

mtyazici committed:
Create two new models named customer and vendor

mtyazici committed:
Change routers for new endpoints customer and vendor

mtyazici committed:
Change controllers and dataAcess for new endpoints customer and vendor

mtyazici committed:
Change services for new endpoints vendor and customer

Also changed server.js and some util functions

mtyazici committed:
Add companyName and companyDomainName to vendor Model

mtyazici committed:
Delete Google functionality from Vendors

We don't want vendors to be able sign up or log in with Google.
I changed all necessary files accordingly.

oztasoi commented on backend/models/vendor.js:
```diff
@@ -0,0 +1,20 @@
+
+const Schema = mongoose.Schema;
+
+var customerSchema = new Schema(
```
@mtyazici var vendorSchema = new Schema(

mtyazici replied:
updated :+1:

oztasoi commented on backend/models/vendor.js:
```diff
@@ -0,0 +1,20 @@
+  { collection: "Clients" }
+);
+
+ClientBase.discriminator("Vendor", customerSchema);
```
ClientBase.discriminator("Vendor", vendorSchema);

mtyazici replied:
updated :+1:

oztasoi reviewed (changes requested):
@mtyazici I've commented necessary changes, sorry not to change on-review. 😁

mtyazici committed:
Update return messages of all endpoints

Also updated swagger.json accordingly and applied suggestions from code review about vendor model.

mtyazici commented:
> @mtyazici I've commented necessary changes, sorry not to change on-review. grin

That's fine :smile:  But I think it is better to show changes here so we can commit them right on the website.

oztasoi reviewed (approved):
@mtyazici now it seems just fine 💯  way to go 🚀
